nifi-nar-bundles/nifi-hl7-bundle/nifi-hl7-processors/src/main/java/org/apache/nifi/processors/hl7/ExtractHL7Attributes.java

```diff
@@ -196,8 +196,8 @@ public void process(final InputStream in) throws IOException {
             final Map<String, String> attributes = getAttributes(message, useSegmentNames, parseSegmentFields);
             flowFile = session.putAllAttributes(flowFile, attributes);
             getLogger().debug("Added the following attributes for {}: {}", new Object[]{flowFile, attributes});
-        } catch (final HL7Exception e) {
-            getLogger().error("Failed to extract attributes from {} due to {}", new Object[]{flowFile, e});
+        } catch (final Throwable t) {
```

Comment:
Thanks for taking a look Matt. The two exceptions I was seeing were a java.lang.NullPointerException (when using a non-recognized segment name) and a java.lang.NumberFormatException (when the first field after the segment name had a code such as RE instead of a sequence number). Both of these exceptions were due to how the processor tries to process the HL7 segments and both were causing the processor to backup after enough messages failed to process and were not routed to failure (or anywhere for that matter). I could catch these two exceptions explicitly but I'd prefer that any error (even an error coming from nifi) cause the flowfile to be routed to failure so that it doesn't backup any of the flowfiles that could be processed successfully. I thought catching at the throwable level would work out well since it looks like that's how the PutHDFS processor checks for failure. Do you disagree?